### PR TITLE
Refactor SPMD docs

### PR DESCRIPTION
**Pull request by JackCaoG**
SPMD doc becomes too long and hard to read. I think we should only ask user to read SPMD basics and point them to more advanced topics if they are interested.

### CODE REVIEW

**Review by jonb377 — APPROVED** [review]
Thanks Jack!

File: docs/spmd_basic.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+t1 = torch,.randn(8, 8, 16).to(device)
+t2 = torch.randn(8).to(device)
+
+# First dimension is being replicated.
+xs.mark_sharding(t1, mesh, (None, ‘data’, ‘model’))
+
+# First dimension is being sharded at data dimension.
+xs.mark_sharding(t2, mesh, (‘data’))
+```
+
```
[the remark this answers is not in the document]
Comment (JackCaoG) [reply docs/spmd_basic.md:74]: hmm I run into 
```
AssertionError: Partition spec length (2) should be equal to the input rank (1).
>>> xs.mark_sharding(t2, mesh, (('data', 'model')))
```
Let me look into the correct syntax.. I remembered I saw it somewhere.

[the remark this answers is not in the document]
Comment (jonb377) [reply docs/spmd_basic.md:74]: Ah my bad lol! That should be a tuple within a tuple: `(('data', 'model'),)`

[the remark this answers is not in the document]
Comment (JackCaoG) [reply docs/spmd_basic.md:74]: Ah ok `xs.mark_sharding( t2, mesh, (('data', 'model'),))`

[the remark this answers is not in the document]
Comment (JackCaoG) [reply docs/spmd_basic.md:74]: above example also needs to be `xs.mark_sharding(t2, mesh, ('data',))`

**Pull request merged**